Sources/Rendering/Core/ImageMapper/index.js

```diff
@@ -16,80 +17,191 @@ function vtkImageMapper(publicAPI, model) {
+      model.slice = id;
       changeDetected = true;
     }
-    publicAPI.setZSlice(id);
 
     if (changeDetected && mtime === publicAPI.getMTime()) {
       publicAPI.modified();
     }
   };
 
-  publicAPI.setYSliceIndex = (id) => {
-    const mtime = publicAPI.getMTime();
-    let changeDetected = false;
-    if (model.currentSlicingMode !== SlicingMode.Y) {
-      model.currentSlicingMode = SlicingMode.Y;
-      changeDetected = true;
-    }
-    publicAPI.setYSlice(id);
+  publicAPI.setXSlice = (id) => publicAPI.setSlice(id, SlicingMode.X);
 
-    if (changeDetected && mtime === publicAPI.getMTime()) {
-      publicAPI.modified();
+  publicAPI.setYSlice = (id) => publicAPI.setSlice(id, SlicingMode.Y);
+
+  publicAPI.setZSlice = (id) => publicAPI.setSlice(id, SlicingMode.Z);
+
+  publicAPI.setISlice = (id) => publicAPI.setSlice(id, SlicingMode.I);
+
+  publicAPI.setJSlice = (id) => publicAPI.setSlice(id, SlicingMode.J);
+
+  publicAPI.setKSlice = (id) => publicAPI.setSlice(id, SlicingMode.K);
+
+  publicAPI.getSlicingModeNormal = () => {
+    const out = [0, 0, 0];
+    const a = publicAPI.getInputData().getDirection();
+    const mat3 = [[a[0], a[1], a[2]], [a[3], a[4], a[5]], [a[6], a[7], a[8]]];
+
+    switch (model.currentSlicingMode) {
+      case SlicingMode.X:
+        out[0] = 1;
+        break;
+      case SlicingMode.Y:
+        out[1] = 1;
+        break;
+      case SlicingMode.Z:
+        out[2] = 1;
+        break;
+      case SlicingMode.I:
+        vtkMath.multiply3x3_vect3(mat3, [1, 0, 0], out);
+        break;
+      case SlicingMode.J:
+        vtkMath.multiply3x3_vect3(mat3, [0, 1, 0], out);
+        break;
+      case SlicingMode.K:
+        vtkMath.multiply3x3_vect3(mat3, [0, 0, 1], out);
+        break;
+      default:
+        break;
     }
+    return out;
   };
 
-  publicAPI.setXSliceIndex = (id) => {
-    const mtime = publicAPI.getMTime();
-    let changeDetected = false;
-    if (model.currentSlicingMode !== SlicingMode.X) {
-      model.currentSlicingMode = SlicingMode.X;
-      changeDetected = true;
+  publicAPI.getClosestIJKAxis = (t = 0.99) => {
+    let inVec3;
+    switch (model.currentSlicingMode) {
+      case SlicingMode.X:
+        inVec3 = [1, 0, 0];
+        break;
+      case SlicingMode.Y:
+        inVec3 = [0, 1, 0];
+        break;
+      case SlicingMode.Z:
+        inVec3 = [0, 0, 1];
+        break;
+      default:
+        return {
+          ijkMode: model.currentSlicingMode,
+          flip: false,
+        };
     }
-    publicAPI.setXSlice(id);
 
-    if (changeDetected && mtime === publicAPI.getMTime()) {
-      publicAPI.modified();
+    // Project vec3 onto direction cosines
+    const out = [0, 0, 0];
+    const a = publicAPI.getInputData().getDirection();
+    const mat3 = [[a[0], a[1], a[2]], [a[3], a[4], a[5]], [a[6], a[7], a[8]]];
+    const mat3I = [[0, 0, 0], [0, 0, 0], [0, 0, 0]];
+    vtkMath.invertMatrix(mat3, mat3I, 3);
+    vtkMath.multiply3x3_vect3(mat3I, inVec3, out);
```

Comment:
Maybe just reorder indicies as rotation inverse is just a transpose I think

Reply:
https://en.wikipedia.org/wiki/Rotation_matrix#Multiplication :raised_hands:

Reply:
Done: https://github.com/Kitware/vtk-js/pull/591/commits/cbdae0dd024f42e5bd75a4b9316d092a8bd38cf0#diff-9762803aa1d1e7c8613c71d155ed4e0dR89